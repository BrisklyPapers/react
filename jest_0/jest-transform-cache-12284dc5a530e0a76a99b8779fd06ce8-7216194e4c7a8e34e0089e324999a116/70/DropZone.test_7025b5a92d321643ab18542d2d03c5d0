'use strict';var _react = require('react');var _react2 = _interopRequireDefault(_react);
var _DropZone = require('../../src/containers/DropZone');var _DropZone2 = _interopRequireDefault(_DropZone);
var _enzyme = require('enzyme');
var _sinon = require('sinon');
var _reduxMockStore = require('redux-mock-store');var _reduxMockStore2 = _interopRequireDefault(_reduxMockStore);
var _reactTestRenderer = require('react-test-renderer');var _reactTestRenderer2 = _interopRequireDefault(_reactTestRenderer);
var _shallow = require('react-test-renderer/shallow');var _shallow2 = _interopRequireDefault(_shallow);function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}
var mockStore = (0, _reduxMockStore2.default)();

describe('containers/DropZone', function () {
    var wrapper = void 0,store = void 0;

    beforeEach(function () {
        store = mockStore({ uploading: false, upload_error: { error: false, errorMessage: "" } });
        store.dispatch = (0, _sinon.spy)();
        wrapper = (0, _enzyme.shallow)(_react2.default.createElement(_DropZone2.default, { store: store }));
    });

    it('renders correctly', function () {
        var renderer = new _shallow2.default();
        renderer.render(
        _react2.default.createElement(_DropZone2.default, { store: store }));

        expect(renderer.getRenderOutput()).toMatchSnapshot();
    });

    it('maps state and dispatch to props', function () {
        expect(wrapper.props()).toEqual(expect.objectContaining({
            uploading: expect.any(Boolean),
            error: expect.any(Boolean),
            errorMessage: expect.any(String),
            storeDocuments: expect.any(Function),
            dropFiles: expect.any(Function) }));

    });

});