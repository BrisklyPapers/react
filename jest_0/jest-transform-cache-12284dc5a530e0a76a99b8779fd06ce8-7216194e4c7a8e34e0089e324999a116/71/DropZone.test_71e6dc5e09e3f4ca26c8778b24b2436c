'use strict';var _react = require('react');var _react2 = _interopRequireDefault(_react);
var _Search = require('../../src/containers/Search');var _Search2 = _interopRequireDefault(_Search);
var _actions = require('../../src/actions');
var _enzyme = require('enzyme');
var _sinon = require('sinon');
var _reduxMockStore = require('redux-mock-store');var _reduxMockStore2 = _interopRequireDefault(_reduxMockStore);function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}

var mockStore = (0, _reduxMockStore2.default)();

describe('containers/Search', function () {
    var wrapper = void 0,store = void 0;

    beforeEach(function () {
        store = mockStore({ searchedText: "fun" });
        store.dispatch = (0, _sinon.spy)();
        wrapper = (0, _enzyme.shallow)(_react2.default.createElement(_Search2.default, { store: store }));
    });

    it('maps state and dispatch to props', function () {
        expect(wrapper.props()).toEqual(expect.objectContaining({
            searchedText: "fun",
            triggerSearch: expect.any(Function),
            onFocus: expect.any(Function) }));

    });

    it('maps triggerSearch to dispatch search action', function () {
        var event = { target: { value: "test" } };
        wrapper.props().triggerSearch(event);

        expect(store.dispatch.callCount).toEqual(2);
        expect(store.dispatch.calledWithMatch({ text: "test", type: _actions.SEARCH_INPUT_KEY_PRESSED })).toEqual(true);
        //expect(store.dispatch.calledWithMatch({text: "test", type: 'SEARCH_DOCUMENTS'})).toEqual(true);
    });

    it('maps triggerSearch to dispatch search action', function () {
        wrapper.props().onFocus();

        expect(store.dispatch.callCount).toEqual(1);
        expect(store.dispatch.calledWithMatch({ type: _actions.PAGEVIEW_SEARCH_RESULTS })).toEqual(true);
    });

});