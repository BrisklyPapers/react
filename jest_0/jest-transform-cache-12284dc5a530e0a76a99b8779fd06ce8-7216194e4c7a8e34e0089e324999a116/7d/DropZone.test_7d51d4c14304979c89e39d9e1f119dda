'use strict';var _react = require('react');var _react2 = _interopRequireDefault(_react);
var _DropZone = require('../../src/containers/DropZone');var _DropZone2 = _interopRequireDefault(_DropZone);
var _actions = require('../../src/actions');
var _enzyme = require('enzyme');
var _sinon = require('sinon');
var _reduxMockStore = require('redux-mock-store');var _reduxMockStore2 = _interopRequireDefault(_reduxMockStore);
var _reactTestRenderer = require('react-test-renderer');var _reactTestRenderer2 = _interopRequireDefault(_reactTestRenderer);
var _shallow = require('react-test-renderer/shallow');var _shallow2 = _interopRequireDefault(_shallow);
var _MuiThemeProvider = require('material-ui/styles/MuiThemeProvider');var _MuiThemeProvider2 = _interopRequireDefault(_MuiThemeProvider);

var _reactTapEventPlugin = require('react-tap-event-plugin');var _reactTapEventPlugin2 = _interopRequireDefault(_reactTapEventPlugin);function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}
(0, _reactTapEventPlugin2.default)();

var mockStore = (0, _reduxMockStore2.default)();

describe('containers/DropZone', function () {
    var wrapper = void 0,store = void 0;

    beforeEach(function () {
        store = mockStore({ uploading: false, upload_error: { error: false, errorMessage: "" } });
        store.dispatch = (0, _sinon.spy)();
    });

    describe('default rendering', function () {
        it('renders correctly', function () {
            var renderer = new _shallow2.default();
            renderer.render(
            _react2.default.createElement(_DropZone2.default, { store: store }));

            expect(renderer.getRenderOutput()).toMatchSnapshot();
        });

        it('maps state and dispatch to props', function () {
            wrapper = (0, _enzyme.shallow)(_react2.default.createElement(_DropZone2.default, { store: store }));
            expect(wrapper.props()).toEqual(expect.objectContaining({
                uploading: expect.any(Boolean),
                error: expect.any(Boolean),
                errorMessage: expect.any(String),
                storeDocuments: expect.any(Function),
                dropFiles: expect.any(Function) }));

        });
    });

    describe('rendering', function () {

        beforeEach(function () {
            wrapper = _reactTestRenderer2.default.create(
            _react2.default.createElement(_MuiThemeProvider2.default, null,
                _react2.default.createElement(_DropZone2.default, { store: store })));


        });

        it('should render all components', function () {
            expect(wrapper.toJSON()).toMatchSnapshot();
        });
    });

    describe('event handlers', function () {

        beforeEach(function () {
            // call shallow twice in order to call inner functions
            // @see https://github.com/reactjs/redux/issues/1534
            wrapper = (0, _enzyme.shallow)(_react2.default.createElement(_DropZone2.default, { store: store })).shallow();
        });

        it('should buffer dropped files', function () {
            var files = [{ file: 'test' }];

            expect(wrapper.state().files).toEqual([]);
            wrapper.instance().onDrop(files);
            expect(wrapper.state().files).toEqual(files);

            var expectedAction = { type: _actions.FILE_DROP, document: { file: 'test' } };
            expect(store.dispatch.calledWithMatch(expectedAction)).toEqual(true);
        });

        var tagReact = { key: 1, label: "react" };
        var tagRedux = { key: 2, label: "redux" };
        var tagJest = { key: 3, label: "jest" };

        it('should add tag to tags', function () {
            expect(wrapper.state().tags).toEqual([]);
            wrapper.instance().addTag(tagReact);
            expect(wrapper.state().tags).toEqual([tagReact]);
            wrapper.instance().addTag(tagRedux);
            expect(wrapper.state().tags).toEqual([tagReact, tagRedux]);
        });

        it('should delete a tag by key', function () {
            expect(wrapper.state().tags).toEqual([]);
            wrapper.instance().addTag(tagReact);
            wrapper.instance().addTag(tagRedux);
            wrapper.instance().addTag(tagJest);
            expect(wrapper.state().tags).toHaveLength(3);

            wrapper.instance().deleteTag(2);
            expect(wrapper.state().tags).toEqual([tagReact, tagJest]);

            wrapper.instance().deleteTag(3);
            expect(wrapper.state().tags).toEqual([tagReact]);

            wrapper.instance().deleteTag(1);
            expect(wrapper.state().tags).toEqual([]);
        });

        it('should not fail if tag to delete does not exist', function () {
            wrapper.instance().deleteTag(1);
            expect(wrapper.state().tags).toEqual([]);

            wrapper.instance().addTag(tagReact);
            wrapper.instance().deleteTag(99);
            expect(wrapper.state().tags).toEqual([tagReact]);
        });

        it('should update current tag', function () {
            expect(wrapper.state().tag).toEqual("");
            wrapper.instance().changeTag("more");
            expect(wrapper.state().tag).toEqual("more");
        });
    });
});


describe('containers/DropZoneComponent', function () {

    it('should trigger storeDocuments', function () {
        var storeDocuments = (0, _sinon.spy)();
        var wrapper = (0, _enzyme.shallow)(_react2.default.createElement(_DropZone.DropZoneComponent, { storeDocuments: storeDocuments })).shallow();
    });
});