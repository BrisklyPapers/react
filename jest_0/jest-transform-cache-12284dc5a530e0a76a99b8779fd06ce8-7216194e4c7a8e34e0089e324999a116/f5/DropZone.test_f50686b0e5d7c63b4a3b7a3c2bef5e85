'use strict';var _react = require('react');var _react2 = _interopRequireDefault(_react);
var _DropZone = require('../../src/containers/DropZone');var _DropZone2 = _interopRequireDefault(_DropZone);
var _enzyme = require('enzyme');
var _sinon = require('sinon');
var _reduxMockStore = require('redux-mock-store');var _reduxMockStore2 = _interopRequireDefault(_reduxMockStore);
var _reactTestRenderer = require('react-test-renderer');var _reactTestRenderer2 = _interopRequireDefault(_reactTestRenderer);
var _shallow = require('react-test-renderer/shallow');var _shallow2 = _interopRequireDefault(_shallow);
var _MuiThemeProvider = require('material-ui/styles/MuiThemeProvider');var _MuiThemeProvider2 = _interopRequireDefault(_MuiThemeProvider);

var _reactTapEventPlugin = require('react-tap-event-plugin');var _reactTapEventPlugin2 = _interopRequireDefault(_reactTapEventPlugin);function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}
(0, _reactTapEventPlugin2.default)();

var mockStore = (0, _reduxMockStore2.default)();

describe('containers/DropZone', function () {
    var wrapper = void 0,store = void 0;

    beforeEach(function () {
        store = mockStore({ uploading: false, upload_error: { error: false, errorMessage: "" } });
        store.dispatch = (0, _sinon.spy)();
        wrapper = (0, _enzyme.shallow)(_react2.default.createElement(_DropZone2.default, { store: store }));
    });

    it('renders correctly', function () {
        var renderer = new _shallow2.default();
        renderer.render(
        _react2.default.createElement(_DropZone2.default, { store: store }));

        expect(renderer.getRenderOutput()).toMatchSnapshot();
    });

    it('maps state and dispatch to props', function () {
        expect(wrapper.props()).toEqual(expect.objectContaining({
            uploading: expect.any(Boolean),
            error: expect.any(Boolean),
            errorMessage: expect.any(String),
            storeDocuments: expect.any(Function),
            dropFiles: expect.any(Function) }));

    });

    it('should initialize default state', function () {
        wrapper = (0, _enzyme.mount)(
        _react2.default.createElement(_MuiThemeProvider2.default, null,
            _react2.default.createElement(_DropZone2.default, { store: store })));



        expect(wrapper.props().childrem[0].store.getState()).toEqual({ tag: "", tags: [], files: [] });
    });

});