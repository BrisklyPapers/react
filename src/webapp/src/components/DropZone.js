import React from 'react';
import Radium from 'radium';
import FileDrop from '../containers/FileDrop';
import InputTag from '../containers/InputTag';
import RaisedButton from 'material-ui/RaisedButton';
import PropTypes from 'prop-types';

class DropZone extends React.Component {

    constructor(props) {
        super(props);
        this.state = {
            tags: [],
            files: []
        }
    };

    render() {
        return (
            <div>
                <FileDrop
                    storeCallback={ this.storeFiles }
                />
                <InputTag
                    storeCallback={ this.storeTags }
                />
                <RaisedButton
                    label="Upload"
                    primary={true}
                    style={styles.button}
                    onClick={this.uploadFiles}
                />
            </div>
        );
    };

    storeTags = (tags) => {
        this.setState({tags: tags});
    };

    storeFiles = (files) => {
        this.setState({files: files});
    };

    uploadFiles = () => {
        this.props.storeDocuments(this.state.files, this.state.tags);
    }
}

DropZone.propTypes = {
    storeDocuments: PropTypes.func.isRequired
};

var styles = {
    button: {
        marginTop: 12,
    }
};

DropZone = Radium(DropZone);

export default DropZone;