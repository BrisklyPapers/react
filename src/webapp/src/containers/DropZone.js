import React from 'react';
import Radium from 'radium';
import FileDrop from '../components/FileDrop';
import InputTag from '../components/InputTag';
import RaisedButton from 'material-ui/RaisedButton';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import {storeDocuments, fileDropped} from '../actions';

class DropZone extends React.Component {

    constructor(props) {
        super(props);
        this.state = this.getInitialState()
    };

    getInitialState = () => {
        return {
            tag: "",
            tags: [],
            files: []
        };
    };

    render() {
        return (
            <div>
                <FileDrop
                    dropFiles={ this.onDrop }
                    files={this.state.files}
                />
                <InputTag
                    addTag={ this.addTag }
                    deleteTag={ this.deleteTag }
                    changeTag={ this.changeTag }
                    tags={this.state.tags}
                    tag={this.state.tag}
                />
                <RaisedButton
                    label="Upload"
                    primary={true}
                    style={styles.button}
                    onClick={this.uploadFiles}
                />
            </div>
        );
    };

    onDrop = (files) => {
        let cFiles = this.state.files;
        for (var i = 0; i < files.length; i++) {
            cFiles.push(files[i]);
        }
        this.setState({files: cFiles});
        this.props.dropFiles(files);
    };

    addTag = (tag) => {
        let tags = this.state.tags;
        tags.push(tag);
        this.setState({tags: tags});
    };

    deleteTag = (key) => {
        let tags = this.state.tags;
        const tagToDelete = tags.map((tag) => tag.key).indexOf(key);
        tags.splice(tagToDelete, 1);
        this.setState({tags: tags});
    };

    changeTag = (value) => {
        this.setState({tag: value});
    };

    uploadFiles = () => {
        this.props.storeDocuments(this.state.files, this.state.tags);
        this.setState(this.getInitialState());
    }
}

DropZone.propTypes = {
    storeDocuments: PropTypes.func.isRequired
};

var styles = {
    button: {
        marginTop: 12,
    }
};

DropZone = Radium(DropZone);

const mapStateToProps = (state) => {
    return {
    };
};

const mapDispatchToProps = (dispatch) => {
    return {
        storeDocuments: (files, tags) => {
            dispatch(storeDocuments(files, tags));
        },
        dropFiles: (files) => {
            for (var i = 0; i < files.length; i++) {
                dispatch(fileDropped(files[i]));
            }
        }
    };
};

DropZone = connect(
    mapStateToProps,
    mapDispatchToProps
)(DropZone);

export default DropZone;