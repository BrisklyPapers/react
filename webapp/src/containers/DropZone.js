import React from 'react';
import Radium from 'radium';
import FileDrop from '../components/FileDrop';
import InputTag from '../components/InputTag';
import RaisedButton from 'material-ui/RaisedButton';
import LinearProgress from 'material-ui/LinearProgress';
import Snackbar from 'material-ui/Snackbar';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import {storeDocuments, fileDropped} from '../actions';

export class DropZoneComponent extends React.Component {

    constructor(props) {
        super(props);
        this.state = this.getInitialState();

        this.onDrop = this.onDrop.bind(this);
        this.addTag = this.addTag.bind(this);
        this.deleteTag = this.deleteTag.bind(this);
        this.changeTag = this.changeTag.bind(this);
        this.uploadFiles = this.uploadFiles.bind(this);
        this.componentWillUpdate = this.componentWillUpdate.bind(this);
    };

    getInitialState() {
        return {
            tag: "",
            tags: [],
            files: [],
            completed: false
        };
    };

    render() {
        return (
            <div>
                <Snackbar
                    open={this.props.error}
                    message={this.props.errorMessage}
                    autoHideDuration={3000}
                    bodyStyle={{backgroundColor: '#D32F2F'}}
                    title="Error"
                />
                <FileDrop
                    dropFiles={ this.onDrop }
                    files={this.state.files}
                />
                <InputTag
                    addTag={ this.addTag }
                    deleteTag={ this.deleteTag }
                    changeTag={ this.changeTag }
                    tags={this.state.tags}
                    tag={this.state.tag}
                />
                {this.props.uploading &&
                <LinearProgress mode="indeterminate" value={this.props.completed}/>
                }
                <RaisedButton
                    label="Upload"
                    primary={true}
                    style={styles.button}
                    onClick={this.uploadFiles}
                    disabled={this.props.uploading}
                  />
            </div>
        );
    };

    onDrop(files) {
        let cFiles = this.state.files;
        for (var i = 0; i < files.length; i++) {
            cFiles.push(files[i]);
        }
        this.setState({files: cFiles});
        this.props.dropFiles(files);
    };

    addTag(tag) {
        let tags = this.state.tags;
        tags.push(tag);
        this.setState({tags: tags});
    };

    deleteTag(key) {
        let tags = this.state.tags;
        const tagToDelete = tags.map((tag) => tag.key).indexOf(key);
        if(0 <= tagToDelete) {
            tags.splice(tagToDelete, 1);
            this.setState({tags: tags});
        }
    };

    changeTag(value) {
        this.setState({tag: value});
    };

    uploadFiles() {
        if (this.state.files.length) {
            this.props.storeDocuments(this.state.files, this.state.tags);
        }
    };

    componentWillUpdate(nextProps, nextState) {
        if (nextProps.uploading === false && nextProps.error === false && this.props.uploading === true) {
            this.setState(this.getInitialState());
        }
    };
}

DropZoneComponent.propTypes = {
    storeDocuments: PropTypes.func.isRequired,
    dropFiles: PropTypes.func.isRequired,
    uploading: PropTypes.bool.isRequired,
    error: PropTypes.bool.isRequired,
    errorMessage: PropTypes.string.isRequired
};

var styles = {
    button: {
        marginTop: 12,
    },
    card: {
        color: "red",
    }
};

DropZoneComponent = Radium(DropZoneComponent);

const mapStateToProps = (state) => {
    return {
        uploading: state.uploading,
        error: state.upload_error.error,
        errorMessage: state.upload_error.errorMessage,
    };
};

const mapDispatchToProps = (dispatch) => {
    return {
        storeDocuments: (files, tags) => {
            dispatch(storeDocuments(files, tags));
        },
        dropFiles: (files) => {
            for (var i = 0; i < files.length; i++) {
                dispatch(fileDropped(files[i]));
            }
        }
    };
};

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(DropZoneComponent);