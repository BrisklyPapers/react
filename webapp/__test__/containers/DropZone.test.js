import React from 'react';
import DropZone from '../../src/containers/DropZone';
import {shallow, mount} from 'enzyme';
import {spy} from 'sinon';
import configureMockStore from 'redux-mock-store';
import renderer from 'react-test-renderer';
import ShallowRenderer from 'react-test-renderer/shallow';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';

import injectTapEventPlugin from 'react-tap-event-plugin';
injectTapEventPlugin();

const mockStore = configureMockStore();

describe('containers/DropZone', () => {
    let wrapper, store;

    beforeEach(() => {
        store = mockStore({uploading: false, upload_error: {error: false, errorMessage: ""}});
        store.dispatch = spy();
    });

    describe('default rendering', () => {
        it('renders correctly', () => {
            const renderer = new ShallowRenderer();
            renderer.render(
                <DropZone store={store} />
            );
            expect(renderer.getRenderOutput()).toMatchSnapshot();
        });

        it('maps state and dispatch to props', () => {
            wrapper = shallow(<DropZone store={store} />);
            expect(wrapper.props()).toEqual(expect.objectContaining ({
                uploading: expect.any(Boolean),
                error: expect.any(Boolean),
                errorMessage: expect.any(String),
                storeDocuments: expect.any(Function),
                dropFiles: expect.any(Function)
            }));
        });
    });

    describe('event handlers', () => {

        beforeEach(() => {
            wrapper = renderer.create(
                <MuiThemeProvider>
                    <DropZone store={store} />
                </MuiThemeProvider>
            );
        });

        it('should render all components', () => {
            expect(wrapper.toJSON()).toMatchSnapshot();
        });

        it('should buffer dropped files', () => {

        });
    });
});
