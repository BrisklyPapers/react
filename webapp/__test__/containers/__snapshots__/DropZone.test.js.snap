// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/DropZone default rendering renders correctly 1`] = `
<DropZone
  dropFiles={[Function]}
  error={false}
  errorMessage=""
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeDocuments={[Function]}
  storeSubscription={
    Subscription {
      "listeners": Object {
        "notify": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": null,
    }
  }
  uploading={false}
/>
`;

exports[`containers/DropZone event handlers should render all components 1`] = `
<div
  data-radium={true}
>
  <div
    open={false}
    style={
      Object {
        "bottom": 0,
        "display": "flex",
        "left": "50%",
        "muiPrepared": true,
        "position": "fixed",
        "transform": "translate(-50%, 48px)",
        "transition": "transform 400ms cubic-bezier(0.23, 1, 0.32, 1) 0ms, visibility 400ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
        "visibility": "hidden",
        "zIndex": 2900,
      }
    }
    title="Error"
  >
    <div
      style={
        Object {
          "backgroundColor": "#D32F2F",
          "borderRadius": 2,
          "flexGrow": 0,
          "fontFamily": "Roboto, sans-serif",
          "height": 48,
          "lineHeight": "48px",
          "maxWidth": 568,
          "minWidth": 288,
          "muiPrepared": true,
          "padding": "0 24px",
          "width": "auto",
        }
      }
      width={3}
    >
      <div
        style={
          Object {
            "color": "#ffffff",
            "fontSize": 14,
            "muiPrepared": true,
            "opacity": 0,
            "transition": "opacity 400ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          }
        }
      >
        <span>
          
        </span>
      </div>
    </div>
  </div>
  <div
    data-radium={true}
  >
    <div
      data-radium={true}
      id="fileDrop"
      onClick={[Function]}
      onDragEnd={[Function]}
      onDragLeave={[Function]}
      onDragOver={[Function]}
      onDrop={[Function]}
      style={
        Object {
          "border": "1px dashed #ccc",
          "minHeight": "100px",
          "padding": "10px",
        }
      }
    >
      Drop files here or click to 
      <span
        id="browse"
      >
        browse
      </span>
      .
      <input
        data-radium={true}
        id="fileBox"
        multiple={true}
        name="files[]"
        onChange={[Function]}
        style={
          Object {
            "display": "none",
          }
        }
        type="file"
      />
    </div>
    <ul
      data-radium={true}
    />
  </div>
  <div
    data-radium={true}
    style={
      Object {
        "display": "flex",
        "flexWrap": "wrap",
      }
    }
  >
    <div
      className={undefined}
      style={
        Object {
          "backgroundColor": "transparent",
          "cursor": "auto",
          "display": "inline-block",
          "fontFamily": "Roboto, sans-serif",
          "fontSize": 16,
          "height": "42px",
          "lineHeight": "24px",
          "muiPrepared": true,
          "position": "relative",
          "transition": "height 200ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "width": 256,
        }
      }
    >
      <div
        style={
          Object {
            "bottom": 12,
            "color": "rgba(0, 0, 0, 0.3)",
            "muiPrepared": true,
            "opacity": 1,
            "position": "absolute",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          }
        }
      >
        Enter tag, press Enter
      </div>
      <input
        disabled={false}
        id="search"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyPress={[Function]}
        style={
          Object {
            "WebkitOpacity": 1,
            "WebkitTapHighlightColor": "rgba(0,0,0,0)",
            "appearance": "textfield",
            "backgroundColor": "rgba(0,0,0,0)",
            "border": "none",
            "color": "rgba(0, 0, 0, 0.87)",
            "cursor": "inherit",
            "font": "inherit",
            "height": "100%",
            "muiPrepared": true,
            "outline": "none",
            "padding": 0,
            "position": "relative",
            "width": "100%",
          }
        }
        type="text"
        value=""
      />
      <div>
        <hr
          aria-hidden="true"
          style={
            Object {
              "borderBottomStyle": "solid",
              "borderBottomWidth": 1,
              "borderColor": "#e0e0e0",
              "borderLeft": "none",
              "borderRight": "none",
              "borderTop": "none",
              "bottom": 8,
              "boxSizing": "content-box",
              "margin": 0,
              "muiPrepared": true,
              "position": "absolute",
              "width": "100%",
            }
          }
        />
        <hr
          aria-hidden="true"
          style={
            Object {
              "borderBottomStyle": "solid",
              "borderBottomWidth": 2,
              "borderColor": "#00bcd4",
              "borderLeft": "none",
              "borderRight": "none",
              "borderTop": "none",
              "bottom": 8,
              "boxSizing": "content-box",
              "margin": 0,
              "muiPrepared": true,
              "position": "absolute",
              "transform": "scaleX(0)",
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "width": "100%",
            }
          }
        />
      </div>
    </div>
  </div>
  <div
    className={undefined}
    style={
      Object {
        "WebkitTapHighlightColor": "rgba(0,0,0,0)",
        "backgroundColor": "#ffffff",
        "borderRadius": 2,
        "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
               0 1px 4px rgba(0, 0, 0, 0.12)",
        "boxSizing": "border-box",
        "color": "rgba(0, 0, 0, 0.87)",
        "display": "inline-block",
        "fontFamily": "Roboto, sans-serif",
        "marginTop": 12,
        "minWidth": 88,
        "muiPrepared": true,
        "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
      }
    }
  >
    <button
      disabled={false}
      href={undefined}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      onTouchTap={[Function]}
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
          "backgroundColor": "#00bcd4",
          "border": 10,
          "borderRadius": 2,
          "boxSizing": "border-box",
          "cursor": "pointer",
          "display": "inline-block",
          "fontFamily": "Roboto, sans-serif",
          "fontSize": "inherit",
          "fontWeight": "inherit",
          "height": 36,
          "lineHeight": "36px",
          "margin": 0,
          "muiPrepared": true,
          "outline": "none",
          "padding": 0,
          "position": "relative",
          "textAlign": "center",
          "textDecoration": "none",
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "verticalAlign": null,
          "width": "100%",
          "zIndex": 1,
        }
      }
      tabIndex={0}
      type="button"
    >
      <div
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
      >
        <div
          style={
            Object {
              "backgroundColor": false,
              "borderRadius": 2,
              "height": 36,
              "muiPrepared": true,
              "top": 0,
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            }
          }
        >
          <span
            style={
              Object {
                "color": "#ffffff",
                "fontSize": 14,
                "fontWeight": 500,
                "letterSpacing": 0,
                "margin": 0,
                "muiPrepared": true,
                "opacity": 1,
                "paddingLeft": 16,
                "paddingRight": 16,
                "position": "relative",
                "textTransform": "uppercase",
                "userSelect": "none",
              }
            }
          >
            Upload
          </span>
        </div>
      </div>
    </button>
  </div>
</div>
`;
